subxt/src/rpc/rpc.rs

```diff
@@ -56,25 +56,28 @@ use codec::{
     Encode,
 };
 use frame_metadata::RuntimeMetadataPrefixed;
+use primitive_types::U256;
 use serde::{
     Deserialize,
     Serialize,
 };
-use sp_core::{
-    storage::{
-        StorageChangeSet,
-        StorageData,
-        StorageKey,
-    },
-    Bytes,
-    U256,
-};
-use sp_runtime::ApplyExtrinsicResult;
 use std::{
     collections::HashMap,
     sync::Arc,
 };
 
+/// Signal what the result of doing a dry run of an extrinsic is.
+pub type DryRunResult = Result<(), DryRunError>;
+
+/// An error dry running an extrinsic.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum DryRunError {
+    /// The extrinsic will not be included in the block
+    TransactionValidityError,
```

Comment:
it's a bummer not to get back the reason why the transaction was rejected... but perhaps a PITA to get

Reply:
Yeah I agree; what I wasn't sure about when I did this was whether anybody is currently using the full details. The reason I didn't include them all right here was just because it involved bringing a load more types over, including DispatchError, which we already have to handle when submitting transactions and we have special logic to decode it from the metadata, so having a "hard copy" felt wrong.

But maybe we do need more details to be returned here?